### PR TITLE
feat: allow rectangular shaped support surfaces

### PR DESCRIPTION
**Pull request by asalzburger**
This PR adds the possibility to let the `SupportHelper` create planar support surfaces.

It also adds more tests and cross-checks for consistent geometry description.

### CODE REVIEW

File: Core/src/Detector/detail/SupportHelper.cpp [hunk 1]
```diff
@@ -215,9 +236,50 @@ void Acts::Experimental::detail::SupportHelper::addSupport(
...
+      throw std::runtime_error(
+          "SupportHelper::addSupport(...) - x,y or z are not constrained, "
+          "needed for rectangular support.");
+    }
+
+    // Get the main support parameters:
+    // - doff .. offset (in r.z)
+    // - dl0min, dl0max .. envelop min, max (in z,r)
+    // - dl1min, dl1max .. envelop min, max (in phi)
+    auto [doff, dl0min, dl0max, dl1min, dl1max] = layerSupportValues;
```
Comment (noemina) [inline Core/src/Detector/detail/SupportHelper.cpp:251]: I guess this, and the calculations below, assumes that the envelopes are all positive numbers. Is this check performed somewhere, or also negative values are possible?

**Review by noemina — COMMENTED** [review]
Looks good to me. I have only a minor comment.

**Comment by paulgessinger** [comment]
@asalzburger will you update this?

File: Tests/UnitTests/Core/Detector/SupportHelperTests.cpp [hunk 2]
```diff
@@ -83,14 +83,24 @@ BOOST_AUTO_TEST_CASE(DiscSupport) {
...
 }
 
+BOOST_AUTO_TEST_CASE(RectangularSupport) {
+  // As a single rectangle
+  auto rectangleSupport =
+      Acts::Experimental::detail::SupportHelper::rectangularSupport(
+          Acts::Transform3::Identity(), {100., 400});
+  BOOST_CHECK(rectangleSupport.size() == 1u);
+  BOOST_CHECK(rectangleSupport[0u]->type() ==
+              Acts::Surface::SurfaceType::Plane);
```
Comment (AJPfleger) [inline Tests/UnitTests/Core/Detector/SupportHelperTests.cpp:93]:
```suggestion
  BOOST_CHECK_EQUAL(rectangleSupport.size(), 1u);
  BOOST_CHECK_EQUAL(rectangleSupport[0u]->type(),
              Acts::Surface::SurfaceType::Plane);
```

File: Tests/UnitTests/Core/Detector/SupportHelperTests.cpp [hunk 3]
```diff
@@ -117,6 +127,29 @@ BOOST_AUTO_TEST_CASE(addCylinderSupport) {
...
+  // Special test to check that support can be smaller than extent
+  // Clear up for the next test
+  sValues = {10, -5, -5, 0., 0.};
+  lSurfaces.clear();
+  assignToAll.clear();
+  // Add split surfaces as support to already exisint surfaces
+  Acts::Experimental::detail::SupportHelper::addSupport(
+      lSurfaces, assignToAll, lExtent, Acts::Surface::SurfaceType::Cylinder,
+      sValues, std::nullopt, 1u);
+  BOOST_CHECK(lSurfaces.size() == 1u);
```
Comment (AJPfleger) [inline Tests/UnitTests/Core/Detector/SupportHelperTests.cpp:140]:
```suggestion
  BOOST_CHECK_EQUAL(lSurfaces.size(), 1u);
```

File: Tests/UnitTests/Core/Detector/SupportHelperTests.cpp [hunk 4]
```diff
@@ -154,6 +187,110 @@ BOOST_AUTO_TEST_CASE(addDiscSupport) {
...
+  sValues = {-10, -10, -20, 0., 0.};
+
+  lSurfaces.clear();
+  assignToAll.clear();
+  // Add split surfaces as support disc
+  Acts::Experimental::detail::SupportHelper::addSupport(
+      lSurfaces, assignToAll, lExtent, Acts::Surface::SurfaceType::Disc,
+      sValues, std::nullopt, 1u);
+
+  BOOST_CHECK(lSurfaces.size() == 1u);
```
Comment (AJPfleger) [inline Tests/UnitTests/Core/Detector/SupportHelperTests.cpp:202]:
```suggestion
  BOOST_CHECK_EQUAL(lSurfaces.size(), 1u);
```

File: Tests/UnitTests/Core/Detector/SupportHelperTests.cpp [hunk 5]
```diff
@@ -154,6 +187,110 @@ BOOST_AUTO_TEST_CASE(addDiscSupport) {
...
+    std::array<Acts::ActsScalar, 5u> sValues = {-10, 10, 10, 20., 20.};
+
+    // Add a single rectangle plane as a support rectantle
+    Acts::Experimental::detail::SupportHelper::addSupport(
+        lSurfaces, assignToAll, lExtent, Acts::Surface::SurfaceType::Plane,
+        sValues, std::nullopt, 1u, sBinning);
+
+    BOOST_CHECK(lSurfaces.size() == 1u);
+    BOOST_CHECK(assignToAll.size() == 1u);
+    BOOST_CHECK(assignToAll[0u] == 0u);
```
Comment (AJPfleger) [inline Tests/UnitTests/Core/Detector/SupportHelperTests.cpp:248]:
```suggestion
    BOOST_CHECK_EQUAL(lSurfaces.size(), 1u);
    BOOST_CHECK_EQUAL(assignToAll.size(), 1u);
    BOOST_CHECK_EQUAL(assignToAll[0u], 0u);
```

**Review by AJPfleger — CHANGES_REQUESTED** [review]

File: Core/src/Detector/detail/SupportHelper.cpp (hunk 1 above)
Comment (asalzburger) [reply Core/src/Detector/detail/SupportHelper.cpp:251]: I think I will stick with garbage in garbage out ... but we can add a exception check.

**Comment by asalzburger** [comment]
This one is superseded by #2879

**Pull request closed without merging**